src/core/client/stream/test/comments/permalink/permalinkViewRejectedComment.spec.tsx

```diff
@@ -0,0 +1,82 @@
+import { render, screen, within } from "@testing-library/react";
+import React from "react";
+
+import {
+  CoralContext,
+  CoralContextProvider,
+} from "coral-framework/lib/bootstrap";
+import { AppContainer } from "coral-stream/App";
+
+import create from "./create";
+
+import {
+  commentWithRejectedReply,
+  replyableComment,
+  settings,
+  stories,
+  unreplyableComment,
+} from "../../fixtures";
+
+const story = stories[0];
+
+function customRenderWithContext(context: CoralContext) {
+  render(
+    <CoralContextProvider value={context}>
+      <AppContainer disableListeners />
+    </CoralContextProvider>
+  );
+}
+
+const createTestRenderer = async () => {
+  const resolvers = {
+    Query: {
+      comment: () => commentWithRejectedReply,
+      settings: () => settings,
+      story: () => story,
+    },
+  };
+
+  const { context } = create({
+    resolvers,
+    initLocalState: (localRecord) => {
+      localRecord.setValue(story.id, "storyID");
+      localRecord.setValue(commentWithRejectedReply.id, "commentID");
+    },
+  });
+
+  customRenderWithContext(context);
+
+  const enabledComment = await screen.findByTestId(
+    `comment-${replyableComment.id}`
+  );
+  const disabledComment = await screen.findByTestId(
+    `comment-${unreplyableComment.id}`
+  );
+
+  return {
+    replyableComment: enabledComment,
+    unreplyableComment: disabledComment,
+  };
+};
+
+it("allows replies to comments with canReply = true", async () => {
+  const {
+    replyableComment: replyableCommentElement,
+  } = await createTestRenderer();
+
+  const replyButton = await within(replyableCommentElement).findByTestId(
```

Comment:
Is there a way to find an element by its role (which I agree is better than its test id) but not depend on the arbitrary name of the commenter?

Reply:
It's also possible to match with regex on substrings, like `name: /Reply to comment by/`, but that will match multiple comments. I think it makes sense to search by the full name of the button, including the commenter's name, because that's what we expect the user to see for the name of the button based on the comment fixtures that we've provided to the test.

Reply:
So is it desirable that the test will break if we change the wording of that button?

Reply:
IMO it makes the tests more readable to have the actual text of the button there, and it's a way to test that translations are applied correctly and what we expect the user to see is what they see. It would break if either 1) the text of the button changes, but I think we'd both want to see that change and it would be fast to fix or 2) the username in the fixture was changed, which seems both unlikely and would also be fast to fix.